rdkit/ML/Scoring/Scoring.py

```diff
@@ -14,6 +14,40 @@
 """
 
 import math
+import random  # Does rdkit depend on numpy? => Use np.random instead
+import copy
+from operator import itemgetter
+
+
+def avoid_sorting_artifacts_on_ranking(scores, col, reverse=False):
```

Comment:
The standard in RDkit is to use camel case.

`col` is not just any column, it is the column you exclude, so something like `excludeColumn` (maybe generalize it to `excludeColumns` or the opposite, `includeColumns`). 

The name is not a description of the function (=what it does). It is a description of the use of the function (the function does no ranking). Consider a different name (e.g. `sortByColumns`) and generalise it to list of include and exclude columns. 
```
def sortByColumns(data, includeColumns=None, excludeColumns=None, 
                  reverse=False, shuffle=True):
  data = copy.copy(data)

  ncolumns = len(data[0])
  if includeColumns is not None:
    columns = set(i % ncolumns for i in includeColumns)
  else
    columns = set(range(len(data[0]))
  if excludeColumns is not None:
    columns -= set(i % ncolumns for i in excludeColumns)

  if shuffle:
    random.shuffle(data)
  return sorted(data, key=lambda x: tuple(xi for i, xi in enumerate(x) if i in columns), 
                reverse=reverse)
```